Algorithm.Python/BasicTemplateFuturesConsolidationAlgorithm.py

```diff
@@ -43,20 +43,22 @@ def Initialize(self):
         future = self.AddFuture(Futures.Indices.SP500EMini)
         future.SetFilter(timedelta(0), timedelta(182))
 
-        self._futureContracts = []
+        self.consolidators = dict()
 
     def OnData(self,slice):
-        for chain in slice.FutureChains:
-            for contract in chain.Value:
-                if contract.Symbol not in self._futureContracts:
-                    self._futureContracts.append(contract.Symbol)
-
-                    consolidator = QuoteBarConsolidator(timedelta(minutes=5))
-                    consolidator.DataConsolidated += self.OnDataConsolidated
-                    self.SubscriptionManager.AddConsolidator(contract.Symbol, consolidator)
-
-                    self.Log("Added new consolidator for " + str(contract.Symbol.Value))
+        pass
 
     def OnDataConsolidated(self, sender, quoteBar):
         self.Log("OnDataConsolidated called on " + str(self.Time))
-        self.Log(str(quoteBar))
\ No newline at end of file
+        self.Log(str(quoteBar))
+        
+    def OnSecuritiesChanged(self, changes):
+        for security in changes.AddedSecurities:
+            consolidator = QuoteBarConsolidator(timedelta(minutes=5))
+            consolidator.DataConsolidated += self.OnDataConsolidated
+            self.SubscriptionManager.AddConsolidator(security.Symbol, consolidator)
+            self.consolidators[security.Symbol] = consolidator
+            
+        for security in changes.RemovedSecurities:
+            consolidator = self.consolidators.pop(security.Symbol)
+            self.SubscriptionManager.RemoveConsolidator(security.Symbol, consolidator)
```

Comment:
To avoid memory leaks need to remove event handlers
`consolidator.DataConsolidated -= self.OnDataConsolidated`